content/blog/management-paradigms-for-virtual-machines-running-on-kubernetes.md

```diff
@@ -0,0 +1,377 @@
+  connection: local
+  tasks:
+  - name: Ensure VM name
+    assert:
+      that: vm is defined
+  - name: Create a VM
+    kubevirt.core.kubevirt_vm:
+      state: present
+      name: "{{ vm }}"
+      namespace: hpe-vmi
+      labels:
+        app: my-example-label
+      instancetype:
+        name: u1.medium
+      preference:
+        name: fedora
+      data_volume_templates:
+        - metadata:
+            name: "{{ vm }}-0"
+          spec:
+            sourceRef:
+              kind: DataSource
+              name: fedora
+              namespace: kubevirt-os-images
+            storage:
+              resources:
+                requests:
+                  storage: 64Gi
+      spec:
+        domain:
+          devices:
+            interfaces:
+            - name: mgmt
+              bridge: {}
+        networks:
+        - name: mgmt
+          multus:
+            networkName: mgmt
+        accessCredentials:
+        - sshPublicKey:
+            propagationMethod:
+              qemuGuestAgent:
+                users:
+                - fedora
+            source:
+              secret:
+                secretName: desktop
+        volumes:
+        - cloudInitConfigDrive:
+            userData: |-
+              #cloud-config
+              # The default username is: fedora
+              runcmd:
+                - [ setsebool, -P, 'virt_qemu_ga_manage_ssh', 'on' ]
+          name: cloudinitdisk
+        - dataVolume:
+            name: "{{ vm }}-0"
+          name: "{{ vm }}-0"
+      wait: yes
+```
+
+Many attributes have been hardcoded in this example but it illustrates the similarities of what `virtctl` outputs based on the parameters provided.
+
+Use the playbook to create a VM:
+
+```shell
+ansible-playbook -e vm=my-vm-0 create_vm.yaml
+```
+
+It takes a minute or so for the VM to come up. When the prompt comes back, create a file named “hosts.kubevirt.yaml” (the “kubevirt.yaml” part of the filename is mandatory):
+
+```yaml
+plugin: kubevirt.core.kubevirt
+namespaces:
+  - hpe-vmi
+host_format: "{name}"
+network_name: mgmt
+label_selector: app=my-example-label
+compose:
+  ansible_user: "'fedora'"
+```
+
+It’s now possible to use the KubeVirt inventory plugin to manage the OS and apps in the VM. Let’s see if it connects:
+
+```shell
+ansible -i hosts.kubevirt.yaml -m ping my-vm-0
+my-vm-0 | SUCCESS => {
+    "ansible_facts": {
+        "discovered_interpreter_python": "/usr/bin/python3.13"
+    },
+    "changed": false,
+    "ping": "pong"
+}
+```
+
+At this point it’s possible to manage the VM like any other host provisioned on any kind of server, hypervisor or cloud platform.
+
+# Summary
+
+It doesn’t matter what your distinct VM management workflow looks like, KubeVirt serves all popular patterns. That said, current tools and processes will require an overhaul and why not switch to idempotent VM management through GitOps while transitioning from your legacy hypervisor? That's a topic for another day.
```

Comment:
I'd recommend splitting out the second sentence in to two:

"That said, current tools and processes will require an overhaul. Why not switch to idempotent VM management through GitOps while transitioning from your legacy hypervisor?" 

Perhaps tack on "in the meantime." after "legacy hypervisor"?